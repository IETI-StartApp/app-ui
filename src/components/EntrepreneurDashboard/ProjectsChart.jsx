import * as React from 'react';
import Paper from '@material-ui/core/Paper';
import {
  Chart,
  BarSeries,
  ArgumentAxis,
  ValueAxis,
} from '@devexpress/dx-react-chart-material-ui';
import { Animation } from '@devexpress/dx-react-chart';


export const ProjectsChart = () => {
    const data= [
      {name: "Proyecto 1", financed: 75}, 
      {name: "Proyecto 2", financed: 57}, 
      {name: "Proyecto 3", financed: 67},
      {name: "Proyecto 4", financed: 17},
      {name: "Proyecto 5", financed: 2},
    ];

    return (
      <Paper>
        <Chart
          data={data}
        >
          <ArgumentAxis />
          <ValueAxis />
          <BarSeries
            barWidth={0.3}
            valueField="financed"
            argumentField="name"
            color="#950740"
          />
          <Animation />
        </Chart>
      </Paper>
    );
}