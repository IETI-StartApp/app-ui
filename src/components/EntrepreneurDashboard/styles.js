import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles((theme) => ({
      fab: {
        position: 'fixed',
        bottom: theme.spacing(5),
        right: theme.spacing(5),
        backgroundColor: '#950740',
        color: '#fff'
    },
}));