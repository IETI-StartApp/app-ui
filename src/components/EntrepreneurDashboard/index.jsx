import React from 'react';
import AttachMoneyIcon from '@material-ui/icons/AttachMoney';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import { Dashboard } from '../Dashboard/index';
import { ProjectsChart } from './ProjectsChart';
import PermIdentityIcon from '@material-ui/icons/PermIdentity';
import CheckIcon from '@material-ui/icons/Check';

export const EntrepreneurDashboard = () => {
    const items = [
        { number: "5", description: "Proyectos Creados", icon: <StarBorderIcon fontSize="large" />, path: "/" },
        { number: "4.5M", description: "Capital recaudado", icon: <AttachMoneyIcon fontSize="large" />, path: "/dashboard" },
        { number: "17", description: "Socios", icon: <PermIdentityIcon fontSize="large" />, path: "/" },
        { number: "0", description: "Proyectos exitosos", icon: <CheckIcon fontSize="large" />, path: "/" }
    ];
    const projects = [
        {
            name: "Proyecto 1", 
            image:"https://www.binarias.org/wp-content/uploads/2020/12/tegnologia.jpeg", 
            finance: { value:15000000, valuation:20000000, startDate: "10/03/2021"}
        },
        {
            name: "Proyecto 2", 
            image:"https://s3-us-west-2.amazonaws.com/wp-mpro-blog/wp-content/uploads/2016/03/14133959/s3-blog-tecnologia-como-prioridad-min.png", 
            finance: { value:10000000, valuation:17500000, startDate: "12/02/2021"},
        },
        {
            name: "Proyecto 3", 
            image:"https://thelogisticsworld.com/wp-content/uploads/2020/02/empresario-analizando-informe-financiero-empresa-graficos-realidad-aumentada_34141-360.jpg", 
            finance: { value:8000000, valuation:12000000, startDate: "21/01/2021"},
        },
    ];

    return (
        <Dashboard items={ items } projects={ projects } chart={ <ProjectsChart /> } projectsTitle={ "Mis proyectos" } chartTitle={ "Mis proyectos: porcentaje financiado" } />
    );
}