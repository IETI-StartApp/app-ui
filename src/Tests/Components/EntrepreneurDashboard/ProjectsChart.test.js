import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { ProjectsChart } from '../../../components/EntrepreneurDashboard/ProjectsChart';

describe('Testing Projects Chart', () => {

    const wrapper = shallow(<ProjectsChart />)

    test('should load Projects Chart', () => {
        expect(wrapper).toMatchSnapshot();
    })
})
