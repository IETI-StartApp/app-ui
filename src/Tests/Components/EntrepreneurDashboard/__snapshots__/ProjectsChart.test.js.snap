// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Projects Chart should load Projects Chart 1`] = `
<WithStyles(ForwardRef(Paper))>
  <Component
    data={
      Array [
        Object {
          "financed": 75,
          "name": "Proyecto 1",
        },
        Object {
          "financed": 57,
          "name": "Proyecto 2",
        },
        Object {
          "financed": 67,
          "name": "Proyecto 3",
        },
        Object {
          "financed": 17,
          "name": "Proyecto 4",
        },
        Object {
          "financed": 2,
          "name": "Proyecto 5",
        },
      ]
    }
  >
    <Component />
    <Component />
    <Component
      argumentField="name"
      barWidth={0.3}
      color="#950740"
      valueField="financed"
    />
    <AnimationBase
      duration={1000}
      easing={[Function]}
    />
  </Component>
</WithStyles(ForwardRef(Paper))>
`;
